src/Features/CSharp/Portable/Snippets/CSharpPropSnippetProvider.cs

```diff
@@ -22,16 +23,16 @@ internal sealed class CSharpPropSnippetProvider() : AbstractCSharpAutoPropertySn
 
     public override string Description => FeaturesResources.property_;
 
-    protected override AccessorDeclarationSyntax? GenerateSetAccessorDeclaration(CSharpSyntaxContext syntaxContext, SyntaxGenerator generator)
+    protected override AccessorDeclarationSyntax? GenerateSetAccessorDeclaration(CSharpSyntaxContext syntaxContext, SyntaxGenerator generator, CancellationToken cancellationToken)
     {
         // Having a property with `set` accessor in a readonly struct leads to a compiler error.
         // So if user executes snippet inside a readonly struct the right thing to do is to not generate `set` accessor at all
         if (syntaxContext.ContainingTypeDeclaration is StructDeclarationSyntax structDeclaration &&
-            structDeclaration.Modifiers.Any(SyntaxKind.ReadOnlyKeyword))
+            syntaxContext.SemanticModel.GetDeclaredSymbol(structDeclaration, cancellationToken) is { IsReadOnly: true })
```

Comment:
💡 Can we make it only use semantics if `partial` modifier is present? So the break down is:

1. `readonly` modifier &rarr; `readonly`
2. No `partial` modifier &rarr; not `readonly`
3. Use semantics

Reply:
I don't like this idea because of 2 things:
1. If we deal with modifiers like this we are basically repeating compiler's semantic logic. And I don't thinks that is a good idea. Language rules should live only in the compiler, so let's not repeat ourselves and reuse this logic
2. If a `class`/`struct` doesn't have a `partial` modifier, but there is another source declaration of this type compiler treats them as partial declarations anyway and just reports error about missing `partial` modifier, since it is more likely that user's intent was actually to have a partial type and they just forgot the modifier. I'm gonna add a test for this case